ucp/benchmarks/README.md

```diff
@@ -0,0 +1,117 @@
+
+
+### Multi-node with manual process setup
+
+This is by far the most complicated setup, and should be used only if running on a cluster where SSH isn't possible or not recommened.
+
+Before diving in the details of this setup, the user should know there is a convenience function to generate commands for each of the nodes. To generate that, the user may refer to the SSH section, where we pass `--hosts` to specify server and worker nodes, this time with an added `--print-commands-only` argument. The output of the command will print one line for server and each node with the exact command that should be executed in those nodes.
+
+Now if the user would like to continue and learn the details this setup, the user is required specify the exact amount of workers that will be executed by the cluster, and the index of each node. Unlike the SSH setup, this time we may not use `--hosts` to specify where to launch.
+
+**WARNING: All processes must specify the same values for `--devs`, `--chunks-per-dev`, `--chunk-size`, `--frac-match`, `--iter`, `--warmup-iter`, `--num-workers`.**
+
+
+#### Calculating number of workers
+
+Calculating the number of workers is straightforward but critical to be done right. It can be calculated as follows:
+
+```python
+# num_workers: number of workers, passed as `--num-workers`
+# len(devs): length of the GPU list specified via `--devs`
+# chunks_per_dev: number of chunks (processes) per GPU specified via `--chunks-per-dev`
+# num_worker_nodes: number of nodes that will run workers
+num_workers = len(devs) * chunks_per_dev * num_worker_nodes
+```
+
+In the examples that follow, we will launch 4 GPUs per node, 2 chunks per device and 2 worker nodes.
+
+```python
+# num_workers = len(devs) * chunks_per_dev * num_worker_nodes
+16 = 4 * 2 * 2
+```
+
+#### Server
+
+First, the user must launch the server, which may be in a node that doesn't contain any GPUs, or could be on one of the nodes where workers will run. To do so, two options exist:
+
+##### Address on a file
+
+In this setup, a file that is reachable on all nodes of the cluster must be specified, for example in a network file system. 
+
+```bash
+python -m ucp.benchmarks.cudf_merge --server --devs 0,1,2,3 --chunks-per-dev 2 --chunk-size 1000000 --frac-match 0.5 --iter 10 --warmup-iter 5 --num-workers 16 --server-file /path/to/network/fs/server.json
+```
+
+##### Address on stdout
+
+```bash
+python -m ucp.benchmarks.cudf_merge --server --devs 0,1,2,3 --chunks-per-dev 2 --chunk-size 1000000 --frac-match 0.5 --iter 10 --warmup-iter 5 --num-workers 16
+```
+
+
+#### Workers
+
+Once the server is up and running, workers must be launched on multiple nodes. Each node must execute workers with all options matching other nodes, both workers and server, with the execption of `--node-idx`. The `--node-idx` argument is used as a unique identifier to each worker, so that it may compute the rank for each GPU worker.
```

Comment:
I would have expected that `--rmm-init-pool-size` is not a collective specification? I guess that's minor though.

Reply:
It's not, there are some others too, like `--collect-garbage`, but specifying different options may lead to situations that may be difficult to ensure they indeed work correctly. So for now I think it's best to try and make things homogenous.